src/renderer/components/ft-list-video/ft-list-video.js

```diff
@@ -444,21 +464,55 @@ export default defineComponent({
     this.parseVideoData()
     this.checkIfWatched()
 
-    if (this.useDeArrowTitles && !this.deArrowCache) {
+    if ((this.useDeArrowTitles || this.useDeArrowThumbnails) && !this.deArrowCache) {
       this.fetchDeArrowData()
     }
+
+    if (this.useDeArrowThumbnails && this.deArrowCache && this.deArrowCache?.thumbnail == null) {
+      if (this.debounceGetDeArrowThumbnail == null) {
+        this.debounceGetDeArrowThumbnail = debounce(this.fetchDeArrowThumbnail, 1000)
+      }
+
+      this.debounceGetDeArrowThumbnail()
+    }
   },
   methods: {
+    fetchDeArrowThumbnail: async function() {
+      if (this.thumbnailPreference === 'hidden') { return }
+      const videoId = this.id
+      const deArrowCacheClone = deepCopy(this.deArrowCache)
+
+      const thumbnail = await deArrowThumbnail(videoId, deArrowCacheClone.thumbnailTimestamp)
+      if (thumbnail) {
+        deArrowCacheClone.thumbnail = thumbnail
+        this.$store.commit('addThumbnailToDeArrowCache', deArrowCacheClone)
+      }
```

Comment:
As you only need the clone if a thumbnail was returned, you can more it inside the `if` statement to save cloning it when it doesn't need to be cloned.

```suggestion
      const thumbnail = await deArrowThumbnail(videoId, this.deArrowCache.thumbnailTimestamp)
      if (thumbnail) {
        const deArrowCacheClone = deepCopy(this.deArrowCache)
        deArrowCacheClone.thumbnail = thumbnail
        this.$store.commit('addThumbnailToDeArrowCache', deArrowCacheClone)
      }
```